### PR TITLE
support mcms in forwarder contract config changeset

### PR DESCRIPTION
**Pull request by krehermann**
Adds mcms support to forwarder contract configuration

### CODE REVIEW

File: deployment/keystone/deploy.go [hunk 1]
```diff
@@ -285,55 +276,22 @@ func ConfigureRegistry(ctx context.Context, lggr logger.Logger, req ConfigureCon
...
-		if err != nil {
-			return fmt.Errorf("failed to configure forwarder for chain selector %d: %w", chain.Selector, err)
-		}
-	}
-	return nil
-}
-
 // Depreciated: use changeset.ConfigureOCR3Contract instead
 // ocr3 contract on the registry chain for the wf dons
-func ConfigureOCR3Contract(env *deployment.Environment, chainSel uint64, dons []RegisteredDon, addrBook deployment.AddressBook, cfg *OracleConfigWithSecrets) error {
```
Comment (krehermann) [inline deployment/keystone/deploy.go:349]: fix up old signature; addr book now in env

File: deployment/keystone/deploy.go [hunk 2]
```diff
@@ -285,55 +276,22 @@ func ConfigureRegistry(ctx context.Context, lggr logger.Logger, req ConfigureCon
...
-		},
-		DonInfos: donsResp.donInfos,
+		Changeset: &deployment.ChangesetOutput{}, // no new addresses, proposals etc
+		DonInfos:  donsResp.donInfos,
 	}, nil
 }
 
-// ConfigureForwardContracts configures the forwarder contracts on all chains for the given DONS
-// the address book is required to contain the an address of the deployed forwarder contract for every chain in the environment
-func ConfigureForwardContracts(env *deployment.Environment, dons []RegisteredDon, addrBook deployment.AddressBook) error {
```
Comment (krehermann) [inline deployment/keystone/deploy.go:318]: code mv'ed to forwarder_deployer

File: deployment/keystone/deploy.go [hunk 3]
```diff
@@ -285,55 +276,22 @@ func ConfigureRegistry(ctx context.Context, lggr logger.Logger, req ConfigureCon
 	lggr.Infow("registered DONs", "dons", len(donsResp.donInfos))
 
 	return &ConfigureContractsResponse{
-		Changeset: &deployment.ChangesetOutput{
-			AddressBook: addrBook,
-		},
-		DonInfos: donsResp.donInfos,
+		Changeset: &deployment.ChangesetOutput{}, // no new addresses, proposals etc
```
Comment (krehermann) [inline deployment/keystone/deploy.go:300]: fix up old bug; should only return new addresses

File: deployment/keystone/types.go [hunk 4]
```diff
@@ -229,7 +229,51 @@ type RegisteredDon struct {
...
 }
 
-func (d RegisteredDon) signers(chainFamily string) []common.Address {
+type RegisteredDonConfig struct {
+	Name             string
+	NodeIDs          []string // ids in the offchain client
+	RegistryChainSel uint64
+}
+
+func NewRegisteredDon(env deployment.Environment, cfg RegisteredDonConfig) (*RegisteredDon, error) {
```
Comment (krehermann) [inline deployment/keystone/types.go:239]: so we can create the needed input to ConfigureForwarderContracts for a given env

**Review by bolekk — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Pull request merged**